// Copyright 2021 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";

enum SpannerDatabaseStateEnum {
  SpannerDatabaseStateEnumNO_VALUE_DO_NOT_USE = 0;
  SpannerDatabaseStateEnumCREATING = 1;
  SpannerDatabaseStateEnumREADY = 2;
}

message SpannerDatabase {
  string name = 1;
  string instance = 2;
  SpannerDatabaseStateEnum state = 3;
  string project = 4;
  repeated string ddl = 5;
}

message ApplySpannerDatabaseRequest {
  SpannerDatabase resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteSpannerDatabaseRequest {
  string service_account_file = 1;
  SpannerDatabase resource = 2;
}

message ListSpannerDatabaseRequest {
  string service_account_file = 1;
  string Project = 2;
  string Instance = 3;
}

message ListSpannerDatabaseResponse {
  repeated SpannerDatabase items = 1;
}

service SpannerDatabaseService {
  rpc ApplySpannerDatabase(ApplySpannerDatabaseRequest) returns (SpannerDatabase);
  rpc DeleteSpannerDatabase(DeleteSpannerDatabaseRequest) returns (google.protobuf.Empty);
  rpc ListSpannerDatabase(ListSpannerDatabaseRequest) returns (ListSpannerDatabaseResponse);
}
